server/src/main/java/org/opensearch/search/approximate/ApproximateScoreQuery.java

```diff
@@ -21,12 +21,12 @@
  * Entry-point for the approximation framework.
  * This class is heavily inspired by {@link org.apache.lucene.search.IndexOrDocValuesQuery}. It acts as a wrapper that consumer two queries, a regular query and an approximate version of the same. By default, it executes the regular query and returns {@link Weight#scorer} for the original query. At run-time, depending on certain constraints, we can re-write the {@code Weight} to use the approximate weight instead.
  */
-public class ApproximateScoreQuery extends Query {
+public final class ApproximateScoreQuery extends Query {
 
     private final Query originalQuery;
     private final ApproximateQuery approximationQuery;
 
-    protected Query resolvedQuery;
+    Query resolvedQuery;
```

Comment:
Do we actually need to change the visibility here?

Reply:
I made `ApproximateScoreQuery` `final`, so IntelliJ gave me a warning, essentially saying "Uhh... `protected` means that subclasses can see it, but this is class is `final` so it can't have subclasses. What are you thinking?"

Reply:
(The value is checked from a unit test, so I couldn't make it `private`. Also, I have a pending change to add another approximate query type in the same package that will read the value from an `ApproximateScoreQuery`, so leaving it package-private is not bad. I would also be open to making it `private` and adding a getter.)